SystemsGenetics/KINC
# 138 cond test mpirun

Description
========
This PR fixes issue #138.   To summarize, there was a bug if the `cond-test` analytic was used with MPI.  All of the clusters would have significant edges. This is because the p-value tests never ran because the class was not properly instantiated for the MPI slaves.  Since the p-value defaults to 0 it looked like everything was significant.  This only affected results if MPI was used.   

To fix this, I updated the initialize function of the ConditionTest class so that the slaves could initialize their instance of the class.   

Also, in the process of debugging this, I fixed a lot of weirdness in the code. It's a bit more streamlined in terms of layout now.  This weirdness did not affect the results. It just made the code harder to figure out. I think it's a bit easier now.  

Testing
=====
To test this PR, do the following prior to pulling this branch:
1.  Run the `example/kinc-gmm-run.sh` script to get correct results.  Save these results in a temp directory for later comparison
2.  Run the `example/kinc-gmm-mpirun.sh` script. You'll see that all of pairs have signficant tests with a p-value of 0.
3. Next pull this branch of the code and re-run step #2.  
4.  Compare the results of step #3 with step #1.  There will be a little bit of discrepancy but not much. The differences are due to issue #137 and all should be fine.

Additional Info
==========
@4ctrl-alt-del or @bentsherman would one of you look over the code to make sure I'm following design standards and I didn't do anything inappropriate with my edits.
